tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -155,7 +163,56 @@ object NativeConfig {
     def withOptimize(value: Boolean): NativeConfig =
       copy(optimize = value)
 
-    override def toString: String =
+    override def withLinktimeProperties(v: Map[String, Any]): NativeConfig = {
+      def isNumberOrString(value: Any) = {
+        def isNonUnitPrimitive = value match {
+          case _: java.lang.Boolean | Boolean => true
+          case _: java.lang.Byte | Byte       => true
+          case _: java.lang.Character | Char  => true
+          case _: java.lang.Short | Short     => true
+          case _: java.lang.Integer | Int     => true
+          case _: java.lang.Long | Long       => true
+          case _: java.lang.Float | Float     => true
+          case _: java.lang.Double | Double   => true
+          case _                              => false
+        }
+
+        def isNonEmptyString = value match {
+          case s: String => s.nonEmpty
+          case _         => false
+        }
```

Comment:
Why reject empty strings?

Reply:
I thought it might be better for the consistency of data, but on the other hand, I can now think of some usages where the empty string would be useful. I'll remove this condition